draft-wang-ppm-differential-privacy.md

```diff
@@ -322,28 +408,42 @@ with the correct noise type.
 
 > TODO: Chris P: we will mention Prio3SumVec because that's what we use to describe aggregator DP with amplification
 
+# Primitives for Implementing DP Mechanisms {#primitives}
 
-# Differential Privacy Mechanism {#dp mechanism}
+This section describes various primitives required for implementing DP
+mechansims. The algorithms are designed to securely expand a short, uniform
+random seed into a sample from a given distribution.
 
-This section should define a framework to apply different dp noise type to a
-client measurement and/or an aggregation.
+## Discrete Laplace
 
-> TODO: Junye to spell out the interface of dp mechanisms, "Noise Mechanisms to Accomplish DP"
+> TODO: Specify a Laplace sampler
+
+## Discrete Gaussian
+
+> TODO: Specify a Gaussian sampler
+
+## Randomized Response
 
+> TODO: Specify any primitives required for randomized response mechanisms
```

Comment:
@junyechen1996 what are examples of these?

Reply:
The earliest paper is probably the one from Google: https://arxiv.org/abs/1407.6981, we haven't cited that one here. [This paper](https://arxiv.org/abs/2211.10082) ({{MJTBp22}}) also talks about two instantiations, one for replacement-DP setting in Appendix A, and one for deletion-DP setting in Appendix C.

Reply:
In your opinion, do any of these algorithms need to be specified? The primary motivation for standardizing Laplace/Gaussian is that there are lots of ways to get the implementation wrong.

I'm familiar with RAPPOR and I'm not sure any algorithm there needs to be standardized.

Reply:
I think it's worth spelling out discrete Gaussian and Laplace. Both are pretty popular options in Aggregator-DP setting.

For RAPPOR, I think having one as an example for Client-DP setting would be useful in the draft.

Reply:
OK, SGTM!